### ISSUE TITLE
Can't add owncloud account

### ISSUE THREAD

**Issue by Sh1d0w** [jangernert/FeedReader#203]
I am using version 1.6.2 of Feedreader. I had previously owncloud acc set but I removed it. After that I am trying to add a new owncloud acc to get RSS from, but the Feedreader client is stuck at the credentials page after I click login.

**Comment by jangernert** [comment]
Feedreader-daemon logs?

**Comment by Sh1d0w** [comment]
@jangernert Sorry, here is the output when I run `feedreader --debug`

```
[ INFO  ] Everything is logged. Used to debug!
[ INFO  ] FeedReader 1.6.2
[ DEBUG ] MainWindow: setupCSS
[ DEBUG ] ContentPage: setup FeedList
[ DEBUG ] UpdateButton: update status
[ DEBUG ] UpdateButton: setSensitive true
[ DEBUG ] MainWindow: determining state
[ DEBUG ] SELECT count(*) FROM main.articles
[ DEBUG ] count 0
[ DEBUG ] SELECT count(*) FROM main.categories
[ DEBUG ] count 0
[ DEBUG ] SELECT count(*) FROM main.feeds
[ DEBUG ] count 0
[ DEBUG ] SELECT count(*) FROM main.tags
[ DEBUG ] count 0
[ DEBUG ] MainWindow: show login

** (feedreader:4183): CRITICAL **: secret_schema_unref: assertion 'g_atomic_int_get (&schema->reserved) > 0' failed

** (feedreader:4183): CRITICAL **: secret_schema_unref: assertion 'g_atomic_int_get (&schema->reserved) > 0' failed
[ DEBUG ] MainWindow: show error bar - errorCode = 6
[ DEBUG ] MainWindow: dont show error bar
[ DEBUG ] HeaderBar: updatebutton status false
[ DEBUG ] UpdateButton: setSensitive false
[ DEBUG ] write login data

** (feedreader:4183): CRITICAL **: file /builddir/build/BUILD/FeedReader-1.6.2/src/Widgets/LoginPage.vala: line 515: uncaught error: Timeout was reached (g-io-error-quark, 24)
[ DEBUG ] DBusConnection: setOnline
[ DEBUG ] FeedList: new FeedList
[ DEBUG ] initCollapseCategories
[ DEBUG ] FeedList: restore selected row: feed -4
[ DEBUG ] ArticleList: newList
[ DEBUG ] ArticleList: create
[ DEBUG ] ArticleList: offset 0
[ DEBUG ] limit: 20
[ DEBUG ] load articles from db
[ DEBUG ] SELECT ROWID, feedID, articleID, title, author, url, preview, unread, marked, tags, date, guidHash FROM main.articles ORDER BY rowid DESC LIMIT 20 OFFSET 0
[ DEBUG ] actual articles loaded: 0
[ DEBUG ] DBusConnection: setOffline
[ DEBUG ] FeedList: new FeedList
[ DEBUG ] initCollapseCategories
[ DEBUG ] FeedList: restore selected row: feed -4
[ DEBUG ] ArticleList: insert new rows
[ DEBUG ] ArticleList: create finished

(feedreader:4183): Gtk-CRITICAL **: gtk_widget_event: assertion 'WIDGET_REALIZED_FOR_EVENT (widget, event)' failed

(feedreader:4183): Gtk-CRITICAL **: gtk_widget_event: assertion 'WIDGET_REALIZED_FOR_EVENT (widget, event)' failed
[ DEBUG ] DBusConnection: springCleanStarted
[ DEBUG ] MainWindow: show springClean
[ DEBUG ] HeaderBar: updatebutton status false
[ DEBUG ] UpdateButton: setSensitive false
[ DEBUG ] DBusConnection: springCleanFinished
[ DEBUG ] MainWindow: show content
[ DEBUG ] FeedList: new FeedList
[ DEBUG ] initCollapseCategories
[ DEBUG ] FeedList: restore selected row: feed -4
[ DEBUG ] HeaderBar: updatebutton status true
[ DEBUG ] UpdateButton: setSensitive true
[ DEBUG ] DBusConnection: syncStarted
[ DEBUG ] ArticleList: get State
[ DEBUG ] scrollpos 0.000000
[ DEBUG ] scrollpos 0.000000
[ DEBUG ] offset 0
[ DEBUG ] DBusConnection: setOffline
[ DEBUG ] FeedList: new FeedList
[ DEBUG ] initCollapseCategories
[ DEBUG ] FeedList: restore selected row: feed -4
[ DEBUG ] DBusConnection: syncFinished
[ DEBUG ] MainWindow: show content
[ DEBUG ] HeaderBar: updatebutton status true
[ DEBUG ] UpdateButton: setSensitive true
[ DEBUG ] UpdateButton: update status
[ DEBUG ] UpdateButton: setSensitive true
[ DEBUG ] ArticleView: size changed
[ DEBUG ] UpdateButton: update status
[ DEBUG ] UpdateButton: setSensitive false
[ DEBUG ] DBusConnection: syncStarted
[ DEBUG ] ArticleList: get State
[ DEBUG ] scrollpos 0.000000
[ DEBUG ] scrollpos 0.000000
[ DEBUG ] offset 0
[ DEBUG ] DBusConnection: setOffline
[ DEBUG ] FeedList: new FeedList
[ DEBUG ] initCollapseCategories
[ DEBUG ] FeedList: restore selected row: feed -4
[ DEBUG ] DBusConnection: syncFinished
[ DEBUG ] MainWindow: show content
[ DEBUG ] HeaderBar: updatebutton status true
[ DEBUG ] UpdateButton: setSensitive true
[ DEBUG ] UpdateButton: update status
[ DEBUG ] UpdateButton: setSensitive true
[ DEBUG ] UpdateButton: update status
[ DEBUG ] UpdateButton: setSensitive false
```

After that it hangs forever on the credentials page so I killed it with CTRL+C after one hour.

**Comment by jangernert** [comment]
Um, No. The --debug argument is not meant for this.
I need the logs of the daemon, not the logs from the UI.

**Comment by Sh1d0w** [comment]
@jangernert Sure, just tell me where I can locate them and I will provide.

**Comment by jangernert** [comment]
Kill the feedreader-daemon and start it from the terminal.

**Comment by Sh1d0w** [comment]
Hey @jangernert here is the log from the daemon:

```
[ INFO  ] Everything is logged. Used to debug!
[ INFO  ] daemon: Notification actions supported
[ INFO  ] FeedReader Daemon 1.6.2
[ DEBUG ] daemon: constructor
[ DEBUG ] daemon: new FeedServer and login
[ DEBUG ] FeedServer: new with type -1
[ DEBUG ] daemon: login status = FEED_READER_LOGIN_RESPONSE_NO_BACKEND
[ DEBUG ] daemon: bus aquired
[ DEBUG ] Daemon: checkOnlineAsync
[ DEBUG ] Daemon: checkOnline
[ DEBUG ] FeedServer: setOffline
[ DEBUG ] daemon: new FeedServer and login
[ DEBUG ] FeedServer: new with type 2
[ DEBUG ] OwnCloud: login
[ DEBUG ] OwnCloud URL: https://cloud.vitanov.org/index.php/apps/news/api/v1-2/
[ ERROR ] ownCloud Message: No response - status code: Connection terminated unexpectedly
[ DEBUG ] FeedServer: setOffline
[ DEBUG ] daemon: login status = FEED_READER_LOGIN_RESPONSE_NO_CONNECTION
```

Could it be that for some reason since I changed the A record for my domain, feedreader could not find the new IP? This shouldn't be the case since from the same machine I am opening it ok in the browser?

**Comment by Sh1d0w** [comment]
Btw I am using latest version of https://github.com/nextcloud/news in my Nextcloud server

**Comment by jangernert** [comment]
Do you know if there have been any API-changes to nextcloud I am not aware of? Afaik there have been none.
Regarding your error:

> https://cloud.vitanov.org/index.php/apps/news/api/v1-2/

is the correct URL?
All I can see is that libsoup didn't get any response to the sent message.

**Comment by Sh1d0w** [comment]
No I don't think there any breaking changes. In the plugins home page it
says it supports Nextcloud 10.

Yes this is the correct domain name. However I don't know about the url
part. All I have to enter in the owncloud account page when adding account
is the OC server, username and password.

Are you able to access the home page of the Nextcloud? If yes then maybe it
is some kind of misconfiguration on my end, despite the fact the plugin is
running ok in the browser.

Will dig tonight on my end, but I was wondering if it is bug on Feedreader
end.

Thanks for the time, will post my findings soon.

Sent from my smartphone. Please excuse any typos.

On Sep 11, 2016 18:58, "Jan Lukas Gernert" <notifications@github.com> wrote:

Do you know if there have been any API-changes to nextcloud I am not aware
of? Afaik there have been none.
Regarding your error:

https://cloud.vitanov.org/index.php/apps/news/api/v1-2/

is the correct URL?
All I can see is that libsoup didn't get any response to the sent message.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/jangernert/FeedReader/issues/203#issuecomment-246187695>,
or mute the thread

**Comment by Sh1d0w** [comment]
@jangernert In the official news documentation I see that the endpoints Feedreader tries to call has been deprecated:

https://github.com/nextcloud/news/blob/master/docs/externalapi/Legacy.md

The new API that is in use is v2:

https://github.com/nextcloud/news/blob/master/docs/externalapi/External-Api.md

I guess this is the problem, since I am using the latest version and respectively API.

**Comment by Sh1d0w** [comment]
The problem was resolved when I deleted the stored credentials in the keychain and re - entered them.

Thanks once again for your time and the great app :)

**Comment by jangernert** [comment]
Afaik the v2 API is still in the planing phase.
Resetting the password is indeed something that is incomplete in the stable version (only ttrss was implemented). This is already fixed in version 2.0 (still in development). Sorry for the inconvenience =/
